geonode/layers/api/serializers.py

```diff
@@ -213,3 +213,43 @@ class DatasetMetadataSerializer(serializers.Serializer):
 
     class Meta:
         fields = "metadata_file"
+
+
+class DatasetTimeSeriesSerializer(serializers.Serializer):
+
+    def __init__(self, *args, **kwargs):
+
+        super().__init__(*args, **kwargs)
+
+        choices = self._get_choices()
+        self.fields["attribute"].choices = choices
+        self.fields["end_attribute"].choices = choices
+
+    @staticmethod
+    def _get_choices():
+
+        attributes = Attribute.objects.all()
+
+        return [(None, "-----")] + [
+            (_a.pk, _a.attribute) for _a in attributes if _a.attribute_type in ["xsd:dateTime", "xsd:date"]
+        ]
```

Comment:
@Gpetrak I would move this inside the Dataset model. The Serializer can call a method on the model to obtain the list of suitable attributes

Reply:
@giohappy done